aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -400,7 +400,7 @@ namespace impl {
     static_assert(std::is_same<guts::typelist::typelist<ParameterTypes...>, typename guts::infer_function_traits_t<KernelFunctor>::parameter_types>::value,
       "Parameter types mismatch");
 
-    static ReturnType call(OperatorKernel* functor, ParameterTypes... args) {
+    static ReturnType call(ParameterTypes... args, OperatorKernel* functor) {
```

Comment:
More for my understanding than anything else- how does this help out the compiler?

Reply:
Function arguments go in fixed places defined by [the System V x86_64 ABI](https://uclibc.org/docs/psABI-x86_64.pdf), at least on Linux & Mac. You can see Low Level System Interface > Function Calling Interface for the gory details (in which case you'll also want to refer to [the Itanium C++ ABI](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#functions) to see how C++ things work), but roughly speaking and ignoring a bunch of subtleties and exceptions, the first 8 pointers' worth of arguments go in a fixed set of 8 registers. (I'll call them r0-r7 for simplicity, but those are definitely not their actual names on x86_64.) Supposing that `functor` takes at most 7 arguments, then with `functor` at the front, we have r0 = `functor` and r1-r7 = the first 7 args to `functor`. We then need to shift r1-r7 into r0-r6 before jumping to `functor`. With `functor` at the back, we have r0-r6 = the first 7 args to `functor` and r7 = `functor`, and we can jump straight to `functor` without moving more things around. 

(If functor takes 8 arguments or more, then we still have the args to it in r0-r7 and the remaining arguments on the stack. Arguments that go on the stack are pushed from right to left and thus popped from left to right, so I haven't checked the assembly for this case but I think that we could just leave `functor` in place, load it from the stack and call it, and then pop it off the stack when `functor` returns.)